Python client contributing (#1228)
G-Research/armada: client/python/README.md

Armada Python Client
=
<hr />

## How To Use This Application
***

> > **Do Not Fork** this application is meant to be used from **[armada_client](https://pypi.org/project/armada-client/)**

Reviewer: Alternatively, this line could be completely removed.

```suggestion
These instructions are intended for end-users. If you wish to develop against armada_client, please see our [documentation on contributing](CONTRIBUTING.md)
```

(and obviously if you took this suggestion, you'd remove the contributing link below.

Reviewer: @kindyluv doesn't look like this comment has been addressed

1) Click on [armada_client](https://pypi.org/project/armada-client/).
2) Read the file [CONTRIBUTING.md](CONTRIBUTING.md)
3) Click [here](https://armadaproject.io/libraries) get an overview of Armada client libraries

Python client wrapping the gRPC services defined in `submit.proto` and `events.proto`; allows for

- submitting, cancelling, and re-prioritising jobs, and
- watching for job events.

## Build
Prerequisites:

For building the python client:

> In the repo level, run `make python` then `cd client/python` and lastly run `poetry install` to pull all the necessary dependencies.

Reviewer: ```suggestion
In the repo level, run `make python` then `cd client/python` and lastly run `pip install .` to install the client from git with all necessary dependencies.
```

Armada Python Client
=
<hr />

## How To Use This Application
***

> **These instructions are intended for end-users.** If you wish to develop against armada_client, please see our **[documentation on contributing](CONTRIBUTING.md )**

1) Click on [armada_client](https://pypi.org/project/armada-client/).
2) Read the file [CONTRIBUTING.md](CONTRIBUTING.md)
3) Click [here](https://armadaproject.io/libraries) get an overview of Armada client libraries

Python client wrapping the gRPC services defined in `submit.proto` and `events.proto`; allows for

- submitting, cancelling, and re-prioritising jobs, and
- watching for job events.

## Build
Prerequisites:

For building the python client:

> In the repo level, run `make python` then `cd client/python` and lastly run `pip install` to install the client from git with all necessary dependencies.

Reviewer 2: ```suggestion
In the repo level, run `make python` then `cd client/python` and lastly run `pip install .` to install the client from git with all necessary dependencies.
```

Reviewer: @kindyluv if this comment is addressed (you can just hit the commit suggestion button) this PR will become mergeable

1) pyenv
   - Sets up local python environment for supporting multiple python environments
   - Set up a local python 3.9 environment
2) pip
   - Package is defined by pyproject.toml

>    - `pip install` will pull dependencies and install based on pyproject.toml

Reviewer: This needs the `.`, `pip install .`

    - Package is defined by pyproject.toml
    - `poetry install` will pull dependencies and install based on pyproject.toml
3) tox

>     - `poetry run tox -e format` will check formatting/linter for your code according to default black settings and flake8

Reviewer: This whole second will need some de-poetry-ing love. I think you can replace `poetry install` on line 30 with `pip install .`, and just remove the `poetry run` prefix from all these tox commands.

Reviewer: @kindyluv doesn't look like this comment has been addressed
